### PR TITLE
feat: add custom template to group on select with groupBy

### PR DESCRIPTION
**Pull request by nartc**
## Summary

Add ability to use custom template for `group` when `groupBy` is on.

normal:
image
with custom template:
image

## Checklist

- [ ] \*Added unit tests
- [ ] \*Added a code reviewer
- [ ] Added changes to `/projects/swimlane/ngx-ui/CHANGELOG.md` under HEAD (Unreleased)
- [ ] Updated the demo page
- [ ] Included screenshots of visual changes

_\*required_

### CODE REVIEW

**Review by joeveiga — APPROVED** [review]
LGTM

**Comment by nelsonGuti** [comment]
Small comment regarding the `'filterable` input discussed via call

File: projects/swimlane/ngx-ui/src/lib/components/select/select-dropdown.component.html [hunk 1]
```diff
@@ -1,79 +1,96 @@
 <div>
-  <div class="ngx-select-filter" *ngIf="filterable && !tagging">
-    <i class="ngx-icon ngx-search"></i>
+  <div class='ngx-select-filter' *ngIf='filterable && !tagging'>
```
Comment (nelsonGuti) [inline projects/swimlane/ngx-ui/src/lib/components/select/select-dropdown.component.html:2]: It seems the prettier didn't run as we switched to from double quotes to single quotes on several of the files...Can we take a look at this?

File: src/app/forms/selects-page/selects-page.component.html [hunk 2]
```diff
@@ -222,6 +222,39 @@ <h4>Objects with Grouping</h4>
...
+  <ng-template #groupByTemplate let-name="groupName">
+    {{name}}
+    <ng-container [ngSwitch]="name">
+      <ng-container *ngSwitchCase="'IOS'">🔥🔥🔥</ng-container>
+      <ng-container *ngSwitchDefault>🎄🎄🎄</ng-container>
+    </ng-container>
+  </ng-template>
+
+  <app-prism>
+<![CDATA[<ngx-select [groupBy]="'type'">
```
Comment (nelsonGuti) [inline src/app/forms/selects-page/selects-page.component.html:240]: Can we update the mirror code using the `groupByTemplate` input?

**Review by nelsonGuti — APPROVED** [review]
LGTM!

**Pull request merged**